src/plugins/intel_gna/src/transformations/insert_copy_layer.cpp

```diff
@@ -101,10 +100,13 @@ InsertCopyBeforeConcatLayer::InsertCopyBeforeConcatLayer() {
             auto concat_input = concat->get_input_node_shared_ptr(i);
             auto current_node = get_prev_node_skipping_certain(concat_input, is_gna_non_functional_node);
 
-            // Crop -> Concat, Input -> Split -> Concat
+            // Crop -> Concat, Input -> Split -> Concat, Memory -> Concat, Parameter -> Concat, Const -> Concat
```

Comment:
Consider adding the purpose description to this comment and move it before the current `for` block (and remove the misleading one which is already there).

Consider the comment like:

Insert copy operation btw the operations including:
    - Parameter - because its memory is allocated in the INPUT memory region,
    - Crop, Split, VariadicSplit - because only the slice of the original buffer should be copied
    - ReadValue - because its memory is allocated in the STATES memory region,
    - Constant - because its memory is allocated in the RO memory region,
and the Concat operation, as its memory is allocated in the SCRATCH memory region.

Reply:
I'll put it to the header

Reply:
Added